galley/pkg/config/analysis/msg/messages.yaml

```diff
@@ -69,3 +69,14 @@ messages:
         type: string
       - name: namespace
         type: string
+
+  - name: "GatewayPortNotOnWorkload"
```

Comment:
Not a blocker for this PR, but do you think it would make sense to genericize this as a warning-level variant of ReferencedResourceNotFound?

Reply:
I can think of several kinds of "not found".

- We name something, and the name is not found (e.g. a Virtual Service uses non-existent subset)
- We have a selector which has no matches (e.g. a Service has no backing pods)
- Two different resource are supposed to provide the same something, but don't (this case)
- We name something which refers to a selector which has no instances (a Virtual Service refers to a host or subset with no pods)

It is unclear which of these should be warnings and which should be errors.
For me it makes sense to break up ReferencedResourceNotFound in a way that helps the user understand how to fix it.  It may be that all or some of these cases deserve variants for error/warning/info.